include/mbedtls/pk.h

```diff
@@ -118,42 +132,108 @@ typedef struct
 #define MBEDTLS_PK_DEBUG_MAX_ITEMS 3
 
 /**
- * \brief           Public key information and operations
+ * \brief           Key pair information and operations
  */
 typedef struct mbedtls_pk_info_t mbedtls_pk_info_t;
 
 /**
- * \brief           Public key container
+ * \brief           Key pair container
```

Comment:
Both “public key” and “key pair” are inaccurate. “Key pair” is closer, but the private half of the pair may be missing. Suggestion:

```
\brief Key pair container

A structure of this type contains either a key pair for asymmetric cryptography, or the public half of a key pair.
```

@yanesca Thoughts?

Reply:
I think whoever wrote the comment meant 'A container for public-key cryptography' and not 'Container for the public half of a key pair' by that. 

If we want to be very precise, then I would suggest something like

> \brief Context structure for public-key cryptographic operations.
>
> This structure contains all the information needed for performing a public-key cryptography operation. Depending on the operation and the cryptographic algorithm it can represent either the public key or the private key and possibly both. Also it may contain other implementation specific data.

Now that I think of it it might be better to call it public-key crypto instead of asymmetric crypto in the documentation.  (I guess that `_pk_` in the module name refers to public-key.)

Reply:
Yes, I'm sure “public-key cryptography” was meant, but that's not what it said.

We don't support a private key without a public key, so the structure can represent either a public key or a pair of matching public and private keys.

Reply:
Done